### PR TITLE
Refactor Opni Insights Service

### PR DESCRIPTION
**Pull request by AmartC**
This PR is primarily focused on modifying the areas_of_interest endpoint so it will reduce the immediate number of areas_of_interest that are returned when it receives a Get request. 

It also adds some error handling and reformats the peaks and areas_of_interest endpoints to follow the coding format which is used for all the other endpoints.

Additionally, logic was added to the logs endpoint to return the total number of logs present within the specified time interval when a Get request is sent to the endpoint.

### CODE REVIEW

File: opni-insights-service/app/main.py [hunk 1]
```diff
@@ -449,6 +450,113 @@ async def get_anomalies_breakdown(start_ts, end_ts):
...
+            Iterate over the areas of interest found and create a larger area of interest time interval which will contain
+            time intervals which are within 10 minutes of the previous area of interest time interval. This reduces
+            the outward number of areas of interest time intervals returned.
+            '''
+            for index, row in areas_of_interest_df.iterrows():
+                if current_aoi_start == -1:
+                    current_aoi_start = row["key"]
+                    aoi_start_ts = row["key"]
+                if not pd.isna(row['time_diff']) and row["time_diff"] != MILLISECONDS_MINUTE:
+                    if row["key"] - prev_segment_start > (11 * MILLISECONDS_MINUTE):
```
Comment (tybalex) [inline opni-insights-service/app/main.py:503]: can we add docstring for this magic number `11`?

File: opni-insights-service/app/main.py [hunk 2]
```diff
@@ -622,95 +741,31 @@ async def index_logs(start_ts: int, end_ts: int, scroll_id: Optional[str] = None
         # Bad Request
         logging.error(e)
 
-
 @app.get("/areas_of_interest")
-async def get_areas_of_interest(start_ts: int, end_ts: int):
+async def index_areas_of_interest(start_ts: int, end_ts: int):
```
Comment (tybalex) [inline opni-insights-service/app/main.py:745]: what's the benefit of having another function `index_areas_of_interest()`, which does nothing else than simply calls `get_areas_of_interest()`?

**Review by tybalex — CHANGES_REQUESTED** [review]

File: opni-insights-service/app/main.py (hunk 2 above)
Comment (AmartC) [reply opni-insights-service/app/main.py:745]: It's to follow the consistency of all the other endpoints in the insights service.

File: opni-insights-service/app/main.py (hunk 1 above)
Comment (AmartC) [reply opni-insights-service/app/main.py]: added documentation and made some changes that should make this clearer.

**Review by tybalex — APPROVED** [review]
Look good to me.
An optional suggestion I have: 
we can leave the endpoints api functions in this main script(such as `index_areas_of_interesting()`), and move the function with actual logic -- such as `get_areas_of_interest()` to another script. This is to make the scripts cleaner and easier maintenance, instead of having 800 lines here. Not necessarily in this release though.

**Review by sanjay920 — CHANGES_REQUESTED** [review]
This script is too large. Pls split into multiple files as YB suggested

**Review by sanjay920 — APPROVED** [review]

**Review by tybalex — APPROVED** [review]

**Review by sanjay920 — APPROVED** [review]

**Pull request merged**